spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/dsl/TaskParser.java

```diff
@@ -16,63 +16,344 @@
+				getTokens().raiseException(secondLabel.startPos, DSLMessage.TASK_NO_DOUBLE_LABELS);
+			}
+		}
+		// Handle a split < ... >
+		if (peek(LT)) {
+			LabelledTaskNode node = parseSplit();
+			node.setLabel(label);
+			// is the split part of a flow? "<..> && b"
+			return parseFlow(node);
+		}
+		TaskAppNode app = eatTaskApp(true);
+		app.setLabel(label);
+		// Handle a potential flow "a && b"
+		return parseFlow(app);
+	}
+
+	private FlowNode parseFlow(LabelledTaskNode firstNodeInFlow) {
+		List<LabelledTaskNode> nodes = new ArrayList<>();
+		nodes.add(firstNodeInFlow);
+		while (maybeEat(ANDAND)) {
+			LabelledTaskNode nextNode = parseTaskNode();
+			// If nextNode is a Flow node, merge it with this one
+			if (nextNode instanceof FlowNode) {
+				nodes.addAll(nextNode.getSeries());
+			}
+			else {
+				nodes.add(nextNode);
+			}
+		}
+		FlowNode f = new FlowNode(nodes);
+		f.setLabel(firstNodeInFlow.getLabel());
+		return f;
+	}
+
+	// '<' jobs ['||' jobs]+ '>'
+	private LabelledTaskNode parseSplit() {
+		List<LabelledTaskNode> flows = new ArrayList<>();
+		Token startSplit = eat(LT);
+		flows.add(parseTaskNode());
+		while (maybeEat(OROR)) {
+			flows.add(parseTaskNode());
+		}
+		Token endSplit = eat(GT);
+		return new SplitNode(startSplit.startPos, endSplit.endPos, flows);
+	}
+
+	// App1
+	// App1 0 -> App2
+	// App1 0 -> App2 1 -> App3
+	// App1 'a' -> App2
+	// App1 'a' -> App2 '*' -> App3
+	// App1 --p1=v1
+	// App1 --p1=v1 'foo'-> Bar --p1=v2
+	private TaskAppNode eatTaskApp(boolean transitionsAllowed) {
+		Token name = eat();
+		if (!name.isKind(TokenKind.IDENTIFIER)) {
+			getTokens().raiseException(name.startPos, DSLMessage.EXPECTED_APPNAME,
+					name.data != null ? name.data : new String(name.getKind().tokenChars));
+		}
+		getTokens().checkpoint();
+		ArgumentNode[] arguments = (inAppMode?maybeEatAppArgs():null);
+		if (!inAppMode && peek(DOUBLE_MINUS)) {
+			getTokens().raiseException(peek().startPos,DSLMessage.TASK_ARGUMENTS_NOT_ALLOWED_UNLESS_IN_APP_MODE);
+		}
+		List<TransitionNode> transitions = transitionsAllowed ? maybeEatTransitions() : Collections.emptyList();
+		return new TaskAppNode(name, arguments, transitions);
+	}
+	
 	/**
-	 * Parse a task definition.
+	 * Return an array of {@link ArgumentNode} and advance the token
+	 * position if the next token(s) contain arguments.
+	 * <p>
+	 * Expected format:
+	 * {@code appArguments : DOUBLE_MINUS identifier(name) EQUALS identifier(value)}
 	 *
-	 * @return the AST for the parsed task
-	 * @throws ParseException
+	 * @return array of arguments or {@code null} if the next token(s) do not
+	 * contain arguments
 	 */
-	public AppNode parse() {
-		AppNode ast = eatApp();
-
-		// Check the task name, however it was specified
-		if (ast.getName() != null && !isValidName(ast.getName())) {
-			throw new ParseException(ast.getName(), 0, DSLMessage.ILLEGAL_TASK_NAME, ast.getName());
+	protected ArgumentNode[] maybeEatAppArgs() {
```

Comment:
Done. Given more time should look at whether to remove and use the arg eating code in the superclass.